docs/Using-Primary-Outputs-for-Post-Actions.md

```diff
@@ -0,0 +1,394 @@
+Example:
+```
+dotnet new TestAssets.TemplateWithSourceBasedRenames --firstRename Awesome 
+```
+The primary output returned by API is: `MyAwesomeTestProject.csproj`.
+The final location of the files is: `./MyAwesomeTestProject.csproj`.
+
+
+# Examples
+
+## Example 1 - apply post action to all primary outputs
+The following example restores all the projects mentioned in primary outputs:
+`template.json` definition
+```
+{
+  "author": "Test Asset",
+  "classifications": [ "Test Asset" ],
+  "name": "TemplateWithSourceNameAndCustomSourceAndTargetPaths",
+  "generatorVersions": "[1.0.0.0-*)",
+  "groupIdentity": "TestAssets.TemplateWithSourceNameAndCustomSourceAndTargetPaths",
+  "precedence": "100",
+  "identity": "TestAssets.TemplateWithSourceNameAndCustomSourceAndTargetPaths",
+  "shortName": "TestAssets.TemplateWithSourceNameAndCustomSourceAndTargetPaths",
+  "sourceName": "MyTestProject",
+  "sources": [
+    {
+      "source": "./Src/Custom/Path/",
+      "target": "./Target/Output/"
+    }
+  ],
+  "primaryOutputs": [
+    {
+      "path": "Target/Output/MyTestProject.csproj"        
+    }
+  ],
+  "postActions": [
+    {
+      "description": "Restore NuGet packages required by this project.",
+      "manualInstructions": [
+        { "text": "Run 'dotnet restore'" }
+      ],
+      "actionId": "210D431B-A78B-4D2F-B762-4ED3E3EA9025",
+      "continueOnError": true,
+    }
+  ]
+}
+```
+
+## Example 2 - reference source files in post action arguments
+The following example restores individual files specified in `files` property. Defining them in primary outputs is optional.
+`template.json` definition
+```
+{
+  "author": "Test Asset",
+  "classifications": [ "Test Asset" ],
+  "name": "TemplateWithSourceNameAndCustomSourceAndTargetPaths",
+  "generatorVersions": "[1.0.0.0-*)",
+  "groupIdentity": "TestAssets.TemplateWithSourceNameAndCustomSourceAndTargetPaths",
+  "precedence": "100",
+  "identity": "TestAssets.TemplateWithSourceNameAndCustomSourceAndTargetPaths",
+  "shortName": "TestAssets.TemplateWithSourceNameAndCustomSourceAndTargetPaths",
+  "sourceName": "MyTestProject",
+  "sources": [
+    {
+      "source": "./Src/Custom/Path/",
+      "target": "./Target/Output/"
+    }
+  ],
+
+  "primaryOutputs": [
+    {
+      "path": "Target/Output/MyTestProject.csproj"           
+    }
+  ],
+
+  "postActions": [
+    {
+      "description": "Restore NuGet packages required by this project.",
+      "manualInstructions": [
+        { "text": "Run 'dotnet restore'" }
+      ],
+      "actionId": "210D431B-A78B-4D2F-B762-4ED3E3EA9025",
+      "continueOnError": true,
+      "args": {
+        "files": [ "Src/Custom/Path/MyTestProject.csproj" ]  //source location
+      }
+    }
+  ]
+}
+```
+Supported for:
+- restore NuGet packages (dotnet CLI)
+- add reference to a project file (dotnet CLI)
+
+## Example 3 - using primary output indexes in post action arguments
```

Comment:
I would somewhat prefer that post actions stop doing this and offer alternative ways of doing this(via file name) and we don't document using indexes, but keep backward compatibility...

Reply:
I think as this is the only way for VS we should mention it for now.

Reply:
Do you both agree that this should be removed? If no than ignore this comment.

If yes, then removing this may be better. I don't think this document is targeting VS developers (we can share this with them internally if needed).

Reply:
My opinion that we should keep it, as the 'Post Actions Registry' includes VS managed post actions ('open the file in editor' as example).
But if it is 2 against 1 for removing it, I will remove. I definitely agree that indexes is not a way to go.

Reply:
Also I removed add project(s) to a solution file (dotnet CLI) here to discourage using it (though it is supported). For VS we have no alternatives